### ISSUE TITLE
[0.4.3] restrict-plus-operands ignores typeof control-flow narrowing

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1262]
## Summary

After a typeof guard narrows unknown to number or string, utlint still treats the operand as unknown. This also occurs after an early-return guard. The unguarded unknown case should continue to report.

Verified with @utoo/lint 0.4.3, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. This reproduction does not depend on different allowAny defaults between plugin versions.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.3 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each snippet separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## Reproductions

```tsx
export function example(a:unknown){if(typeof a==='number'){return a+1;}return 0;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `unknown`.
```

```tsx
export function example(a:unknown){if(typeof a!=='number')return 0;return a+1;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `unknown`.
```

```tsx
export function example(a:unknown){if(typeof a==='string')return a+'x';return '';}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `unknown`.
```

## Controls

```tsx
export function example(a:unknown){return a+1;}
```

ESLint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `unknown`.
```

utlint:

```text
Invalid operand for a '+' operation. Operands must each be a number or string. Got `unknown`.
```

```tsx
export function example(a:number,b:number){return a+b;}
```

ESLint:

```text
(no diagnostics)
```

utlint:

```text
(no diagnostics)
```

There are no parsing failures or lint-disable comments, and only this rule is enabled.
